Framework/PythonInterface/test/python/plugins/algorithms/MuonMaxEntTest.py

```diff
@@ -11,51 +11,48 @@
-        inputData = CreateWorkspace(DataX=x_data, DataY=y_data, DataE=e_data, UnitX="Time")
-        return inputData
-
-    def genData2(self):
-        x_data = np.linspace(0, 30.0, 100)
-        e_data = []
-        y_data = []
-        for xx in x_data:
-            y_data.append(np.sin(2.3 * xx + 0.1) * np.exp(-xx / 2.19703))
-            e_data.append(np.cos(0.2 * xx))
-        for xx in x_data:
-            y_data.append(np.sin(4.3 * xx + 0.2) * np.exp(-xx / 2.19703))
-            e_data.append(np.cos(0.2 * xx))
+def create_workspace():
+    x_data = np.linspace(0, 30.0, 100)
+    e_data = []
+    y_data = []
+    for xx in x_data:
+        y_data.append(np.sin(2.3 * xx + 0.1) * np.exp(-xx / 2.19703))
+        e_data.append(np.cos(0.2 * xx))
+    for xx in x_data:
+        y_data.append(np.sin(4.3 * xx + 0.2) * np.exp(-xx / 2.19703))
+        e_data.append(np.cos(0.2 * xx))
+
+    inputData = CreateWorkspace(DataX=x_data, DataY=y_data, DataE=e_data, NSpec=2, UnitX="Time", StoreInADS=False)
+    return inputData
+
+
+def create_workspace_with_dead_detectors():
+    inputData = CreateSimulationWorkspace("MUSR", "0,1,32")
+    xData = (inputData.dataX(0)[1:] + inputData.dataX(0)[:-1]) / 2.0
+    for j in range(inputData.getNumberHistograms()):
+        if j == 42 or j == 24:
+            inputData.dataY(j)[:] = np.zeros(len(xData))
+            inputData.dataE(j)[:] = np.cos(0.3 * xData)
+        else:
+            phi = 2.0 * math.pi * (j + 1) / (inputData.getNumberHistograms() + 1)
+            inputData.dataY(j)[:] = np.sin(2.3 * xData + phi)
+            inputData.dataE(j)[:] = np.cos(0.3 * xData)
+    return inputData
 
-        inputData = CreateWorkspace(DataX=x_data, DataY=y_data, DataE=e_data, NSpec=2, UnitX="Time")
-        return inputData
 
-    def genDataWithDeadDetectors(self):
-        inputData = CreateSimulationWorkspace("MUSR", "0,1,32")
-        xData = (inputData.dataX(0)[1:] + inputData.dataX(0)[:-1]) / 2.0
-        for j in range(inputData.getNumberHistograms()):
-            if j == 42 or j == 24:
-                inputData.dataY(j)[:] = np.zeros(len(xData))
-                inputData.dataE(j)[:] = np.cos(0.3 * xData)
-            else:
-                phi = 2.0 * math.pi * (j + 1) / (inputData.getNumberHistograms() + 1)
-                inputData.dataY(j)[:] = np.sin(2.3 * xData + phi)
-                inputData.dataE(j)[:] = np.cos(0.3 * xData)
-        return inputData
+class MuonMaxEntTest(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        cls._workspace = create_workspace()
+        cls._workspace_with_dead_detectors = create_workspace_with_dead_detectors()
 
-    def cleanUp(self):
-        DeleteWorkspace("InputData")
-        DeleteWorkspace("freq")
-        DeleteWorkspace("time")
-        DeleteWorkspace("phase")
+    @classmethod
+    def tearDownClass(cls) -> None:
+        AnalysisDataService.clear()
 
-    def test_executes(self):
```

Comment:
Are we getting rid of this test because we don't think it adds anything that isn't already covered by `test_executes_on_multiple_spectra`?

Reply:
Thanks for reviewing this, and yes that is correct